components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/OperationPoliciesApiServiceImpl.java

```diff
@@ -0,0 +1,329 @@
+import org.wso2.carbon.apimgt.api.APIProvider;
+import org.wso2.carbon.apimgt.api.ExceptionCodes;
+import org.wso2.carbon.apimgt.api.model.OperationPolicyData;
+import org.wso2.carbon.apimgt.api.model.OperationPolicyDefinition;
+import org.wso2.carbon.apimgt.api.model.OperationPolicySpecification;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.importexport.utils.CommonUtil;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.rest.api.common.RestApiCommonUtil;
+import org.wso2.carbon.apimgt.rest.api.common.RestApiConstants;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.OperationPoliciesApiService;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.common.mappings.OperationPolicyMappingUtil;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.OperationPolicyDataDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.OperationPolicyDataListDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.utils.RestApiPublisherUtils;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+
+import java.io.File;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URLConnection;
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+public class OperationPoliciesApiServiceImpl implements OperationPoliciesApiService {
+
+    private static final Log log = LogFactory.getLog(OperationPoliciesApiServiceImpl.class);
+
+    /**
+     * Add a common operation policy that can be used by all the APIs
+     *
+     * @param policySpecFileInputStream              Input stream of the common policy specification file
+     * @param policySpecFileDetail                   Common policy specification
+     * @param synapsePolicyDefinitionFileInputStream Input stream of the synapse policy definition file
+     * @param synapsePolicyDefinitionFileDetail      Synapse definition of the operation policy
+     * @param ccPolicyDefinitionFileInputStream      Input stream of the choreo connect policy definition file
+     * @param ccPolicyDefinitionFileDetail           Choreo connect definition of the operation policy
+     * @param messageContext                         message context
+     * @return Added common operation policy DTO as response
+     */
+    @Override
+    public Response addCommonOperationPolicy(InputStream policySpecFileInputStream, Attachment policySpecFileDetail,
+                                             InputStream synapsePolicyDefinitionFileInputStream,
+                                             Attachment synapsePolicyDefinitionFileDetail,
+                                             InputStream ccPolicyDefinitionFileInputStream,
+                                             Attachment ccPolicyDefinitionFileDetail,
+                                             MessageContext messageContext) throws APIManagementException {
+
+        try {
+            String jsonContent = "";
```

Comment:
Can move to L85

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/11040/commits/aabf4548185a3c63b94e1943085e1f4d99363450